docs/v6-migration.md

```diff
@@ -0,0 +1,103 @@
+
+There is, however, a new setting introduced into `config.js` named `legacyLinking`. This is an option introduced mainly to accommodate users of qBittorrent who want to utilize the Auto Torrent Management feature. This will link in the flat-folder style previously seen in v5. If you want the old (v5) style behavior of linking or use AutoTMM, then you will need to set this option to `true`.
+
+### torrentDir
+
+Previously, our recommendation if you wanted to strictly search only `dataDirs` for matches was to point `torrentDir` at a empty folder. This is no longer necessary. You can now set `torrentDir` to `null` to achieve a data-only search.
+
+### blockList
+
+We have added a new config option called `blockList`. This is an array (like `torznab` or `dataDirs`) and requires square brackets with each string wrapped with quotes and separated by commas (["example", "example2"]) - these strings can be either a whole torrent/file name, a bit of text contained in the name, or the entire infoHash from the torrent you want to block.
+
+### apiAuth and apiKey
+
+API Authentication for `cross-seed` API endpoints (announce and webhook) is now required, however, as a compromise to those who may be using automated deployment or do not have the desire for randomly generated keys we have an option in `config.js` for a predesignated API key. The previously used `apiAuth` has been deprecated/removed.
+
+Leaving `apiKey` as `undefined` will continue to use a generated key which you can retrieve with `cross-seed api-key` - while providing a specific key will override that generated key.
+
+:::warning
+While we provide this option, we strongly urge you to use a generated key. If you are specifying a key, please make sure this is a secure and safe key. You are responsible for your security.
+:::
+
+### Support for .env file
+
+In v6 we've introduced native support for reading `.env` files. You can use this file to protect your API keys and passwords. Usage of this feature is entirely optional but encouraged if you have the knowledge or experience with this type of configuration.
+
+The `.env` file needs to be present in the **current working directory** when you run cross-seed. In the case of the docker, the `.env` file needs to be inside the `/config` folder for the container to load it for you. For a `systemd` daemon, you set the `WorkingDirectory`.
+
+To access a value that is defined in the `.env` you use `process.env.VALUE`.
+
+Example:
+
+- `/config/.env`
+  ```
+  PROWLARR_API_KEY=abcd
+  QBIT_URL="http://username:password@qbittorrent:8080"
+  ```
+- `/config/config.js`
+  ```
+  module.exports = {
+    // ...
+    torznab: [
+      `http://prowlarr:9696/12/api?apikey=${process.env.PROWLARR_API_KEY}`
+    ],
+    // ...
+    qbittorrentUrl: process.env.QBIT_URL
+    // ...
+  }
+  ```
+
+:::note
+Any string with a variable inside of it should have the `process.env` reference wrapped with `${}` as well as enclosed in backticks (`) instead of quotes.
+:::

```

Comment:
I've noticed a few help requests explicitly mentioning they person isn't technical. I wonder if changing this note a little would help avoid future syntax issues with variables.

```suggestion
:::note
Strings with a variable inside of them should use backticks (\`) instead of quotes around the string. The variable should be wrapped in curly braces (`{}`) and be preceeded with a dollar sign (`$`).

For example:

```js
torznab: [
    "http://prowlarr:9696/12/api?apikey=ThisIsMyNormalApiKey",
    \`http://prowlarr:9696/2/api?apikey=${process.env.MY_PROWLARR_KEY}\`
]`
\```
:::
```
The formatting might be tricky though. 🤔

Reply:
I've changed the formatting a bit, will let @mmgoodnow weigh in before I change the wording since its pretty specific in its usage.

Reply:
how often are people doing things like this? (template strings in the config files) I think I'd rather deemphasize it so that users don't have to deal with more esoteric JS syntax if they don't want to.

Reply:
Well, they aren't now....cause you can't do this lol....

But given that it's possible now, it seems likely that the problem might present itself if we present it this way (template strings)....however, I think the more apt solution would be to put the entire link into the .env, and then label it something like "TRACKER1" so that they can easily identify them when they want to remove or manage them in the array.

Thoughts?

Reply:
I don't think it matters too much, as long as we preface it with "this is an advanced feature"

Reply:
When I push this in a minute or two, you'll see what I propose.

I think it's a better, easier thing to implement. Not everyone who uses env will necessarily be able to, or want to, use the required template string syntax. I don't frankly think its necessary, and as you'll see it will be more manageable in the config file as I say.

I'll leave this unresolved and await your thoughts.